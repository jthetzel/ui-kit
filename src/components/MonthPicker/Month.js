import styled from 'react-emotion';

export default styled.div`
  width: calc((100% / 3) - 20px);
  text-align: center;
  padding: 5px;
  margin: 5px;
  background-color: ${props => (props.isActive ? '#f5f4f5' : '#ffffff')};
  color: ${props => (props.isDisabled ? '#90a4ae' : '#263238')};
  pointer-events: ${props => (props.isDisabled ? 'none' : 'auto')};
  
  
  &:hover {
    cursor: pointer;
    color: #223bf2;
  }
`;
