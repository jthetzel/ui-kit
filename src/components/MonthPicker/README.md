MonthPicker

```js
initialState = {
  date: new Date()
};

const changeMonth = (value) => {
  state.date.setMonth(value);
  setState({ date: new Date(state.date) });
};

<div>
  <div>{state.date.toString()}</div>
  <MonthPicker
    value={state.date.getMonth()}
    onChange={changeMonth}
  />
</div>
```

YearPicker + MonthPicker

```js
initialState = {
  date: new Date(),
  year: new Date().getFullYear()
};
const { date, year } = state;
const activeYear = date.getFullYear() === year;

const changeMonth = (value) => {
  date.setMonth(value);
  date.setFullYear(year);
  setState({ date: new Date(date) });
};

<div>
  <div>{state.date.toString()}</div>
  <YearPicker
    value={year}
    onChange={(value) => setState({ year: value })}
  />
  <MonthPicker
    value={activeYear ? date.getMonth() : null}
    onChange={changeMonth}
  />
</div>
```

YearPicker + MonthPicker (disabled dates)

```js
initialState = {
  date: new Date(),
  year: new Date().getFullYear()
};
const { date, year } = state;
const activeYear = date.getFullYear() === year;

const changeMonth = (value) => {
  date.setMonth(value);
  date.setFullYear(year);
  setState({ date: new Date(date) });
};

const disabled = (month) => {
  if (year > new Date().getFullYear()) {
    return true;
  }
  
  if (year < new Date().getFullYear()) {
    return false;
  }

  return month > new Date().getMonth();
};

<div>
  <div>{state.date.toString()}</div>
  <YearPicker
    value={year}
    onChange={(value) => setState({ year: value })}
  />
  <MonthPicker
    value={activeYear ? date.getMonth() : null}
    onChange={changeMonth}
    disabled={disabled}
  />
</div>
```