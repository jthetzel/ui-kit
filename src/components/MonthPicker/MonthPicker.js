import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Style
import Container from './Container';
import Month from './Month';

class MonthPicker extends Component {
  constructor(props) {
    super(props);

    this.state = {
      date: new Date()
    };

    this.onClickMonth = this.onClickMonth.bind(this);
  }

  onClickMonth(date) {
    this.setState({ date });
    this.props.onChange(date);
  }

  render() {
    const { date } = this.state;
    const { months, disabled, activeYear } = this.props;

    return (
      <Container>
        {
          months.map((month, i) => {
            const isActive = date.getMonth() === i && activeYear;
            const currentDate = new Date(date);
            currentDate.setMonth(i);

            return (
              <Month
                key={month}
                onClick={this.onClickMonth.bind(null, currentDate)}
                isActive={isActive}
                isDisabled={disabled(currentDate)}
              >
                {month}
              </Month>
            );
          })
        }
      </Container>
    );
  }
}

MonthPicker.propTypes = {
  onChange: PropTypes.func,
  activeYear: PropTypes.bool,
  months: PropTypes.arrayOf(PropTypes.string),
  disabled: PropTypes.func
};

MonthPicker.defaultProps = {
  onChange: () => {},
  disabled: () => false,
  activeYear: true,
  months: [
    'Январь',
    'Февраль',
    'Март',
    'Апрель',
    'Май',
    'Июнь',
    'Июль',
    'Август',
    'Сентябрь',
    'Октябрь',
    'Ноябрь',
    'Декабрь'
  ]
};

export default MonthPicker;
