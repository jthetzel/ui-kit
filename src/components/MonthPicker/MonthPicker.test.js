import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MonthPicker from '../MonthPicker';


test('MonthPicker does not crash', () => {
  const wrapper = shallow(<MonthPicker value={0} onChange={() => {}} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('MonthPicker change value', () => {
  const wrapper = mount(<MonthPicker value={0} onChange={() => {}} />);
  wrapper.setProps({ value: 2 });
  expect(wrapper.props().value).toBe(2);
});

test('MonthPicker disabled', () => {
  const wrapper = shallow(<MonthPicker value={0} onChange={() => {}} />);
  wrapper.setProps({ disabled: month => month === 3 });
  expect(toJson(wrapper)).toMatchSnapshot();
});
