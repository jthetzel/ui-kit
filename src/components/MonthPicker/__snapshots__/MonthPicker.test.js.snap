// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker disabled 1`] = `
<Styled(div)>
  <Styled(div)
    isActive={true}
    isDisabled={false}
    key="Январь"
    onClick={[Function]}
  >
    Январь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Февраль"
    onClick={[Function]}
  >
    Февраль
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Март"
    onClick={[Function]}
  >
    Март
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={true}
    key="Апрель"
    onClick={[Function]}
  >
    Апрель
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Май"
    onClick={[Function]}
  >
    Май
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Июнь"
    onClick={[Function]}
  >
    Июнь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Июль"
    onClick={[Function]}
  >
    Июль
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Август"
    onClick={[Function]}
  >
    Август
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Сентябрь"
    onClick={[Function]}
  >
    Сентябрь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Октябрь"
    onClick={[Function]}
  >
    Октябрь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Ноябрь"
    onClick={[Function]}
  >
    Ноябрь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Декабрь"
    onClick={[Function]}
  >
    Декабрь
  </Styled(div)>
</Styled(div)>
`;

exports[`MonthPicker does not crash 1`] = `
<Styled(div)>
  <Styled(div)
    isActive={true}
    isDisabled={false}
    key="Январь"
    onClick={[Function]}
  >
    Январь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Февраль"
    onClick={[Function]}
  >
    Февраль
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Март"
    onClick={[Function]}
  >
    Март
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Апрель"
    onClick={[Function]}
  >
    Апрель
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Май"
    onClick={[Function]}
  >
    Май
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Июнь"
    onClick={[Function]}
  >
    Июнь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Июль"
    onClick={[Function]}
  >
    Июль
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Август"
    onClick={[Function]}
  >
    Август
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Сентябрь"
    onClick={[Function]}
  >
    Сентябрь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Октябрь"
    onClick={[Function]}
  >
    Октябрь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Ноябрь"
    onClick={[Function]}
  >
    Ноябрь
  </Styled(div)>
  <Styled(div)
    isActive={false}
    isDisabled={false}
    key="Декабрь"
    onClick={[Function]}
  >
    Декабрь
  </Styled(div)>
</Styled(div)>
`;
